From 7ac17223e785976f32f8c5d2888ae5d7492d5cbb Mon Sep 17 00:00:00 2001
From: Josh Wilsdon <josh@wilsdon.ca>
Date: Fri, 25 May 2018 11:29:53 -0700
Subject: [PATCH] TRITON-432 cn-agent should ignore non-log entries in
 stdio.log Reviewed by: Richard Kiene <richard.kiene@joyent.com> Approved by:
 Richard Kiene <richard.kiene@joyent.com>

---
 lib/backends/smartos/lib/docker-stdio.js | 30 ++++++++++++++----------
 package.json                             |  2 +-
 2 files changed, 18 insertions(+), 14 deletions(-)

diff --git a/lib/backends/smartos/lib/docker-stdio.js b/lib/backends/smartos/lib/docker-stdio.js
index 9a6289f..bc01407 100644
--- a/lib/backends/smartos/lib/docker-stdio.js
+++ b/lib/backends/smartos/lib/docker-stdio.js
@@ -1304,7 +1304,9 @@ function runContainerLogsCommand(container, params, socket) {
     });
 
     lstream.on('readable', function runContainerLogsCommandOnReadable() {
+        var data;
         var line;
+        var rec;
 
         while ((line = lstream.read()) != null) {
             line = line.trim();
@@ -1318,20 +1320,22 @@ function runContainerLogsCommand(container, params, socket) {
                 break;
             }
 
-            var rec = JSON.parse(line);
-            var data;
-            if (params.Timestamps) {
-                data = rec.time + ' ' + rec.log;
-            } else {
-                data = rec.log;
-            }
+            rec = JSON.parse(line);
+            // only include entries that are stdio logs
+            if (rec.hasOwnProperty('log')) {
+                if (params.Timestamps) {
+                    data = rec.time + ' ' + rec.log;
+                } else {
+                    data = rec.log;
+                }
 
-            writeData({
-                log: log,
-                streamType: rec.stream,
-                stream: socket,
-                chunk: data.toString()
-            });
+                writeData({
+                    log: log,
+                    streamType: rec.stream,
+                    stream: socket,
+                    chunk: data.toString()
+                });
+            }
         }
     });
 
diff --git a/package.json b/package.json
index fe64b85..83bc438 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cn-agent",
   "description": "Triton Compute Node Agent",
-  "version": "2.1.1",
+  "version": "2.1.2",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
-- 
2.21.0

